doc/sequence_diagram.puml

```diff
@@ -5,67 +5,67 @@ skinparam dpi 300
 actor email_alert_service as "email-alert-service"
 participant controller as "ContentChangeController"
 database content_change as "ContentChange"
-participant subscription_fulfilment_worker as "SubscriptionFulfilmentWorker"
+participant subscription_content_worker as "SubscriptionContentWorker"
 participant subscription_matcher as "SubscriptionMatcher"
-database email_entry as "EmailEntry"
-participant email_builder_worker as "EmailBuilderWorker"
+database subscription_content as "SubscriptionContent"
+participant email_generation_worker as "EmailGenerationWorker"
 control digest_timer as "DigestTimer"
 database email as "Email"
-participant email_presenter as "EmailPresenter"
-participant email_delivery_worker as "EmailDeliveryWorker"
+participant email_renderer as "EmailRenderer"
+participant delivery_request_worker as "DeliveryRequestWorker"
 database delivery_attempt as "DeliveryAttempt"
 control delivery_monitor as "DeliveryMonitor"
 participant govuk_notify as "GOV.UK Notify" #orange
 
 email_alert_service -> controller : POST /create
 controller -> content_change : create
 content_change -> controller : id
-controller -> subscription_fulfilment_worker : enqueue with id
+controller -> subscription_content_worker : enqueue with id
 controller -> email_alert_service : acknowledge
 
 group sidekiq
-  subscription_fulfilment_worker -> content_change : find by id
-  content_change -> subscription_fulfilment_worker : instance
-  subscription_fulfilment_worker -> subscription_matcher : content_change
+  subscription_content_worker -> content_change : find by id
+  content_change -> subscription_content_worker : instance
+  subscription_content_worker -> subscription_matcher : content_change
   subscription_matcher -> subscription_matcher : ... existing logic
```

Comment:
Is this the point that we interact with `Subscribable` (`SubscriberList` in the current system), `Subscription` and `Subscriber` models? Do we want to capture this on the sequence diagram?

Reply:
Yes, at this point the `SubscriptionMatcher` matches the `ContentChange` against `Subscribables` then finds all `Subscriptions` associated with these.

We probably don't interact with the `Subscriber` much at this point. We're more interested in their associated `Subscriptions`.

We could add it to the diagram but I decided not to because:
- the diagram's already quite unwieldy and I didn't want to make it bigger
- the matching behaviour's super complicated and I didn't want to unpick it all

I think there's probably a separate task to take a look at that matching logic and simplify it. For example, it's currently using lots of Postgres JSON querying techniques that I think we could get rid of eventually.